src/App.tsx

```diff
@@ -1,14 +1,44 @@
 /* eslint-disable max-len */
-import React from 'react';
+
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loader, setLoader] = useState(true);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [searchInputValue, setSearchInputValue] = useState('');
+
+  useEffect(() => {
+    getTodos().then(fetchedTodos => {
+      setTodos(fetchedTodos);
+      setLoader(false);
+    });
+  }, []);
+
+  const filteredTodos = todos.filter(todo =>
+    todo.title.toLowerCase().includes(searchInputValue.toLowerCase()),
+  ).filter(todo => {
+    switch (selectedOption) {
+      case 'active':
+        return !todo.completed;
+      case 'completed':
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
```

Comment:
The current implementation of filtering todos performs two separate filter operations which could be combined into one for better performance. Consider merging these filters to reduce the number of iterations over the todos array.